src/nvidiautil@ethanwharris/extension.js

```diff
@@ -119,50 +103,78 @@ function disable() {
+
+  icons = icons.concat(logo);
+  labels = labels.concat(label);
+  settings_call += setting;
+  parse_function = and_then(parse_function, parse);
+}
+
+/*
  * Re-Load the settings (use as a callback for settings changes)
  */
 function load_settings() {
-  show_utilisation = extension_settings.get_boolean(SETTINGS_UTILISATION);
-  show_temperature = extension_settings.get_boolean(SETTINGS_TEMPERATURE);
-  show_memory = extension_settings.get_boolean(SETTINGS_MEMORY);
+  var show_utilisation = extension_settings.get_boolean(SETTINGS_UTILISATION);
+  var show_temperature = extension_settings.get_boolean(SETTINGS_TEMPERATURE);
+  var show_memory = extension_settings.get_boolean(SETTINGS_MEMORY);
+
+  settings_call = 'nvidia-settings ';
+  parse_function = function(lines, values) {
+    return values;
+  };
 
   icons = [];
   labels = [];
-  formatters = [];
 
   if(show_utilisation) {
-    var logo_util = new St.Icon({icon_name: 'nvidia-card-symbolic', style_class: 'system-status-icon'});
-    var util_label = new St.Label({text: "", style_class: 'label'});
-
-    icons = icons.concat(logo_util);
-    labels = labels.concat(util_label);
-    formatters = formatters.concat(function(val) {
-      return val + "%";
+    build_property('nvidia-card-symbolic', '-q GPUUtilization ', function(lines, values) {
+      var line = lines.shift();
+      var util = line.substring(9,11);
+      util = util.replace(/\D/g,'');
+      return values.concat(util + "%");
     });
   }
 
   if(show_temperature) {
-    var logo_temp = new St.Icon({icon_name: 'nvidia-temp-symbolic', style_class: 'system-status-icon'});
-    var temp_label = new St.Label({text: "", style_class: 'label'});
-
-    icons = icons.concat(logo_temp);
-    labels = labels.concat(temp_label);
-    formatters = formatters.concat(function(val) {
-      return val + "\xB0" + "C";
+    build_property('nvidia-temp-symbolic', '-q GPUCoreTemp ', function(lines, values) {
+      var temp = lines.shift();
+      lines.shift();
+      return values.concat(temp + "\xB0" + "C");
     });
   }
 
   if(show_memory) {
-    var logo_ram = new St.Icon({icon_name: 'nvidia-ram-symbolic', style_class: 'system-status-icon'});
-    var mem_label = new St.Label ({text: "", style_class: 'label'});
-
-    icons = icons.concat(logo_ram);
-    labels = labels.concat(mem_label);
-    formatters = formatters.concat(function(val) {
-      return val + "%";
+    build_property('nvidia-ram-symbolic', '-q UsedDedicatedGPUMemory -q TotalDedicatedGPUMemory ', function(lines, values) {
+      var used_memory = lines.shift();
+      var total_memory = lines.shift();
+      var mem_usage = ((used_memory / total_memory) * 100).toString();
+      mem_usage = mem_usage.substring(0,2);
+      mem_usage = mem_usage.replace(/\D/g,'');
+      return values.concat(mem_usage + "%");
     });
   }
 
+  settings_call += '-t';
+
+  if (!show_utilisation && !show_temperature && !show_memory) {
+    settings_call = 'echo N/A';
```

Comment:
Could instead turn off the callback - seems a bit redundant to echo NA every two seconds

Reply:
Fixed, now de-activates the timer